File: 492-construct-the-rectangle/constructRectangle.cpp
#include <iostream>
#include <vector>
#include <cmath>
#include <map>

using namespace std;
/*
这题太简单，忽略
*/
vector<int> constructRectangle(int area) {
    vector<int> ret;
    for (int i = sqrt(area); i >= 1; i--) {
        if (area % i != 0) continue;
        ret.push_back(area / i);
        ret.push_back(i);
        break;
    }

    return ret;
}
int main() {
    constructRectangle(4);
    return 0;
}

File: 492-construct-the-rectangle/constructRectangle.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
import math

class Solution(object):
    def constructRectangle(self, area):
        """
        :type area: int
        :rtype: List[int]
        """
        i = int(math.sqrt(area))
        while i > 0:
            if area % i != 0:
                i -= 1
                continue
            return [int(area / i), i]

s = Solution()
print(s.constructRectangle(1))
